pandas/core/categorical.py

```diff
@@ -436,9 +436,12 @@ def astype(self, dtype, copy=True):
 
         """
         if is_categorical_dtype(dtype):
-            if copy is True:
-                return self.copy()
-            return self
+            # GH 10696/18593
+            dtype = self.dtype._update_dtype(dtype)
```

Comment:
we might want to add some explicit tests for ``_update_dtype`` at some point (separate PR)

Reply:
added them in the other PR during the initial implementation actually:

https://github.com/pandas-dev/pandas/blob/265e3272b8709a7be274321ee8b505a0e74b8e10/pandas/tests/dtypes/test_dtypes.py#L127-L152